src/snowflake/snowpark/mock/analyzer.py

```diff
@@ -479,11 +479,7 @@ def unary_expression_extractor(
                 raise NotImplementedError(
                     f"[Local Testing] Expression {type(expr.child).__name__} is not implemented."
                 )
-            expr_str = (
-                expr.name
-                if expr.name
-                else self.analyze(expr.child, expr_to_alias, parse_local_name)
-            )
+            expr_str = self.analyze(expr.child, expr_to_alias, parse_local_name)
```

Comment:
I don't think I understand how this change could enable `substr` 😂

Reply:
This is to make sure the generated column name is something like `SUBSTRING("A", 1, 3)` instead of `SUBSTRING("A")`, so `col("a").substring(1, 3)` and `col("a").substring(2, 100)` won't overwrite each other in the result.